maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/AbstractSurefireMojo.java

```diff
@@ -2542,8 +2563,57 @@ private TestClassPath generateTestClasspath() {
             classpathArtifacts = filterArtifacts(classpathArtifacts, dependencyFilter);
         }
 
+        Set<String> additionalClasspathElements = new HashSet<>();
+        if (getAdditionalClasspathElements() != null) {
+            Arrays.stream(getAdditionalClasspathElements()).forEach(additionalClasspathElements::add);
+        }
+        if (getAdditionalClasspathArtifacts() != null) {
+            resolveDependencies(Arrays.stream(getAdditionalClasspathArtifacts())).stream()
+                    .map(File::getAbsolutePath)
+                    .forEach(additionalClasspathElements::add);
+        }
         return new TestClassPath(
-                classpathArtifacts, getMainBuildPath(), getTestClassesDirectory(), getAdditionalClasspathElements());
+                classpathArtifacts, getMainBuildPath(), getTestClassesDirectory(), additionalClasspathElements);
+    }
+
+    Set<File> resolveDependencies(Stream<String> mavenCoordinates) throws MojoFailureException {
```

Comment:
Should we move this method and `resolveArtifact` to `SurefireDependencyResolver` instead?